### DISCUSSION TITLE
ElementFinder and ElementArrayFinder

### DISCUSSION THREAD

**Discussion by divyagupta31** [webdriverio/codemod#67] [Migration Issue]
how are we supposed to import elementfinder and elementArrayFinder here, like we used to do in protractor?

**Comment by christian-bromann** [comment]
Why do you need to import them?

**Reply by divyagupta31** [reply]
this is how we used it earlier..  i was looking if we follow same approach here?? if not, then what's the way around?

**Reply by christian-bromann** [reply]
If you want to type WebdriverIO elements you can either use `WebdriverIO.Element`, `WebdriverIO.ElementArray` or `WebdriverIO.ChainablePromiseElement`, `WebdriverIO.ChainablePromiseElement`. The `WebdriverIO` namespace is available for you if you set `webdriverio/async` in your types and documented in the TypeScript docs.

**Reply by divyagupta31** [reply]
![image](https://user-images.githubusercontent.com/40352515/188660639-e5e55739-d0f8-4a17-87e3-979795d26b0c.png)
i am facing this issue?

**Reply by christian-bromann** [reply]
not sure how your function looks like but imagine this:

```ts
public async clickOnElement (uiElement: WebdriverIO.Element | WebdriverIO.ChainablePromiseElement) {
    await uiElement.click()
    // ...
}
```

**Reply by divyagupta31** [reply]
![image](https://user-images.githubusercontent.com/40352515/188672810-efa83088-33c9-464c-b7ef-c0503720929b.png). We wanna check if passed argument is single element or list of element, for which we used instanceof
this is how our function look like. not sure what we are missing here, as we are still seeing issues.
1.Cannot used namespace'webdriverIO' as a value
2. Property click does not exist on type(Element|ElementArray)

**Reply by christian-bromann** [reply]
> 1.Cannot used namespace'webdriverIO' as a value

Well this is a type not a value, if you want to check the instance of get the prototype name

> 2. Property click does not exist on type(Element|ElementArray)

not sure why you do `await(uiElement).click()` when `await uiElement.click()` is sufficient.

I recommend to run `npm create wdio` and select a TypeScript setup or check out some of the [boilerplate projects](https://webdriver.io/docs/boilerplates) that have TypeScript in it to see how to set this up.
